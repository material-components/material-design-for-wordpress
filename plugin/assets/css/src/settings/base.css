/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

.material-settings {
	--mdc-typography-headline1-font-family: "Roboto", sans-serif;
	--mdc-typography-headline2-font-family: "Roboto", sans-serif;
	--mdc-typography-headline3-font-family: "Roboto", sans-serif;
	--mdc-typography-headline4-font-family: "Roboto", sans-serif;
	--mdc-typography-headline5-font-family: "Roboto", sans-serif;
	--mdc-typography-headline6-font-family: "Roboto", sans-serif;
	--mdc-typography-subtitle1-font-family: "Roboto", sans-serif;
	--mdc-typography-subtitle2-font-family: "Roboto", sans-serif;
	--mdc-typography-font-family: "Roboto", sans-serif;
	--mdc-typography-body1-font-family: "Roboto", sans-serif;
	--mdc-typography-body2-font-family: "Roboto", sans-serif;
	--mdc-typography-caption-font-family: "Roboto", sans-serif;
	--mdc-typography-button-font-family: "Roboto", sans-serif;
	--mdc-typography-overline-font-family: "Roboto", sans-serif;
	--material-design-settings-background-color: #fff;
	--material-design-settings-primary-color: #212121;
	--material-design-settings-primary-text: #fff;
	--material-design-wizard-primary-color: #212121;
	--material-design-settings-error-color: #dc3232;

	& .mdc-button {
		--mdc-theme-primary: var(--material-design-settings-primary-color);
		--mdc-theme-on-primary: var(--material-design-settings-primary-text);
	}

	& h1,
	& h3,
	& h4 {
		margin-top: 0;
	}

	& p {
		font-size: 16px;
		margin: 0;
	}

	& .material-settings__updates {
		margin-top: 24px;
	}

	& .mdc-button--raised .mdc-circular-progress {
		--mdc-theme-primary: #fff;
	}

	& .mdc-button.is-loading {
		text-align: left;
		width: 105px;
	}

	& .material-settings__message {
		align-items: center;
		color: var(--material-design-settings-error-color);
		display: flex;
	}

	& .material-settings__message-text {
		font-size: 16px;
		font-weight: 500;
		margin-left: 10px;
	}
}
