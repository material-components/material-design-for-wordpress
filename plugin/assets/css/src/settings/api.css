.material-settings__api {
	padding-left: 0;
	padding-right: 0;

	& .components-base-control__field {
		height: 100%;
		margin-bottom: 0;
	}

	& .material-settings__api-input input {
		height: 100%;
		width: 100%;
	}
}
