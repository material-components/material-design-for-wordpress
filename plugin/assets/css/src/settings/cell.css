.material-settings__cell--justify-end {
	justify-self: flex-end;
}
