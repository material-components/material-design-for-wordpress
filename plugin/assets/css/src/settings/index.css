@import "./base.css";
@import "./updater.css";
@import "./switch.css";
@import "./cell.css";
@import "./api.css";
