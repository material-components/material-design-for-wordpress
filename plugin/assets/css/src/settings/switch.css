.material-settings__switch {
	align-items: center;
	display: flex;

	& .mdc-switch {
		margin-right: 15px;
	}
}
