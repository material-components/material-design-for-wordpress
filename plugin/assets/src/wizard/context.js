/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { createContext, useReducer } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { reducer } from './reducer';
import { STEPS, STATUS, ADDONS } from './constants';

/**
 * Setups context to be used across the app
 */
const StepContext = createContext();

const { Provider } = StepContext;

/**
 * Default state of the world
 *
 */
const initialState = {
	addons: Object.keys( ADDONS ),
	active: STEPS.WELCOME,
	previous: [],
	status: STATUS.IDLE,
	error: {},
};

/**
 * Creates a wrapper in order to use our custom reducer
 *
 * @param {*} param Children to render
 */
export const StepProvider = ( { children } ) => {
	const [ state, dispatch ] = useReducer( reducer, initialState );

	return <Provider value={ { state, dispatch } }>{ children }</Provider>;
};

export default StepContext;
