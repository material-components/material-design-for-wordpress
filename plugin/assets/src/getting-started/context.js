/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { useReducer, createContext } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { STATUS } from '../wizard/constants';
import { reducer } from './reducer';
import { TABS } from './constants';

/**
 * Setups context to be used across the app
 */
const TabContext = createContext();

const { Provider } = TabContext;

const tabs = Object.keys( TABS );

/**
 * Default state of the world
 *
 */
const initialState = {
	activeTab: 'CUSTOMIZE',
	completed: [ 'WIZARD' ],
	status: STATUS.IDLE,
	actionToInstall: null,
	error: {},
	themeStatus: 'install',
	contentStatus: 'install',
	tabs,
};

/**
 * Creates a wrapper in order to use our custom reducer
 *
 * @param {*} param Children to render
 */
export const TabProvider = ( { children } ) => {
	const [ state, dispatch ] = useReducer( reducer, initialState );

	return <Provider value={ { state, dispatch } }>{ children }</Provider>;
};

export default TabContext;
