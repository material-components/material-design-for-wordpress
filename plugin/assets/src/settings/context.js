/**
 * WordPress dependencies
 */
import { useReducer, createContext } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { STATUS } from '../wizard/constants';
import { UPDATERS } from './constants';
import { reducer } from './reducer';
import getConfig from '../admin/get-config';

/**
 * Setups context to be used across the app
 */
const SettingsContext = createContext();

const { Provider } = SettingsContext;

const initialState = {
	status: STATUS.IDLE,
	apiStatus: getConfig( 'apiStatus' ),
	updates: {},
	availableUpdates: Object.keys( UPDATERS ),
	updaters: UPDATERS,
	error: {},
};

initialState.updates[ UPDATERS.FONTS.type ] = true;
initialState.updates[ UPDATERS.ICONS.type ] = false;

/**
 * Creates a wrapper in order to use our custom reducer
 *
 * @param {*} param Children to render
 */
export const SettingsProvider = ( { children } ) => {
	const [ state, dispatch ] = useReducer( reducer, initialState );

	return <Provider value={ { state, dispatch } }>{ children }</Provider>;
};

export default SettingsContext;
