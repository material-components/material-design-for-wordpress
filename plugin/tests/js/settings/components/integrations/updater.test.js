/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import Updater from '../../../../../assets/src/settings/components/integrations/updater';
import { SettingsProvider } from '../../../../../assets/src/settings/context';

const baseProps = {
	title: 'Google Fonts',
	type: 'FONTS',
	lastUpdated: parseInt( Date.now(), 10 ),
	needsKey: true,
	checked: true,
	apiStatus: 'ok',
	updateAvailable: true,
};

const setup = props => {
	return render(
		<SettingsProvider>
			<Updater { ...props } />
		</SettingsProvider>
	);
};

describe( 'Settings: Updater', () => {
	it( 'matches snapshot', () => {
		const { container } = setup( baseProps );
		expect( container ).toMatchSnapshot();
	} );

	it( 'matches snapshot when disabled', () => {
		const props = {
			...baseProps,
			apiStatus: 'install',
		};

		const { container } = setup( props );
		expect( container ).toMatchSnapshot();
	} );

	it( 'matches snapshot when never updated', () => {
		const props = {
			...baseProps,
			lastUpdated: null,
		};

		const { container } = setup( props );
		expect( container ).toMatchSnapshot();
	} );

	it( 'matches snapshot when should not update', () => {
		const props = {
			...baseProps,
			updateAvailable: false,
		};

		const { container } = setup( props );
		expect( container ).toMatchSnapshot();
	} );
} );
