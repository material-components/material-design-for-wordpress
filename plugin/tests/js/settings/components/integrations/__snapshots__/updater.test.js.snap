// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings: Updater matches snapshot 1`] = `
<div>
  <div
    class="material-settings__updater"
  >
    <div
      class="mdc-layout-grid"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="mdc-layout-grid__cell mdc-layout-grid__cell--span-7 mdc-layout-grid__cell--align-middle"
        >
          <h3
            class="mdc-typography--headline6"
          >
            Google Fonts
          </h3>
          <p
            class="mdc-typography--body1"
          >
            Last update on Apr 7, 2021, 09:47 PM
          </p>
        </div>
        <div
          class="mdc-layout-grid__cell mdc-layout-grid__cell--span-3 mdc-layout-grid__cell--align-middle material-settings__cell--justify-end"
        >
          <div
            class="material-settings__switch"
          >
            <div
              class="mdc-switch material-wizard-switch mdc-switch--checked"
            >
              <div
                class="mdc-switch__track"
              />
              <div
                class="mdc-switch__thumb-underlay"
              >
                <div
                  class="mdc-switch__thumb"
                />
                <input
                  aria-checked="true"
                  checked=""
                  class="mdc-switch__native-control"
                  id="updater-1"
                  role="switch"
                  type="checkbox"
                />
              </div>
            </div>
            <label
              for="updater-1"
              id="label-updater-1"
            >
              Auto-updates enabled
            </label>
          </div>
        </div>
        <div
          class="mdc-layout-grid__cell mdc-layout-grid__cell--span-2 mdc-layout-grid__cell--align-middle material-settings__cell--justify-end"
        >
          <button
            class="mdc-button mdc-button--raised"
          >
            <i
              aria-hidden="true"
              class="material-icons mdc-button__icon leading-icon"
            >
              cached
            </i>
            <span
              class="mdc-button__label"
            >
              Update
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Settings: Updater matches snapshot when disabled 1`] = `
<div>
  <div
    class="material-settings__updater"
  >
    <div
      class="mdc-layout-grid"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="mdc-layout-grid__cell mdc-layout-grid__cell--span-7 mdc-layout-grid__cell--align-middle"
        >
          <h3
            class="mdc-typography--headline6"
          >
            Google Fonts
          </h3>
          <p
            class="mdc-typography--body1"
          >
            To enable Google Fonts updates please 
            <a
              href="https://developers.google.com/fonts/docs/developer_api#APIKey"
              rel="noopener noreferrer"
              target="_blank"
            >
              activate Google API Key
            </a>
             first
          </p>
        </div>
        <div
          class="mdc-layout-grid__cell mdc-layout-grid__cell--span-3 mdc-layout-grid__cell--align-middle material-settings__cell--justify-end"
        />
        <div
          class="mdc-layout-grid__cell mdc-layout-grid__cell--span-2 mdc-layout-grid__cell--align-middle material-settings__cell--justify-end"
        >
          <div
            class="material-settings__message material-settings__message--error"
          >
            <i
              aria-hidden="true"
              class="material-icons mdc-button__icon leading-icon"
            >
              error
            </i>
            <span
              class="material-settings__message-text"
            >
              Updates disabled
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Settings: Updater matches snapshot when never updated 1`] = `
<div>
  <div
    class="material-settings__updater"
  >
    <div
      class="mdc-layout-grid"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="mdc-layout-grid__cell mdc-layout-grid__cell--span-7 mdc-layout-grid__cell--align-middle"
        >
          <h3
            class="mdc-typography--headline6"
          >
            Google Fonts
          </h3>
          <p
            class="mdc-typography--body1"
          >
            Last update on never
          </p>
        </div>
        <div
          class="mdc-layout-grid__cell mdc-layout-grid__cell--span-3 mdc-layout-grid__cell--align-middle material-settings__cell--justify-end"
        >
          <div
            class="material-settings__switch"
          >
            <div
              class="mdc-switch material-wizard-switch mdc-switch--checked"
            >
              <div
                class="mdc-switch__track"
              />
              <div
                class="mdc-switch__thumb-underlay"
              >
                <div
                  class="mdc-switch__thumb"
                />
                <input
                  aria-checked="true"
                  checked=""
                  class="mdc-switch__native-control"
                  id="updater-3"
                  role="switch"
                  type="checkbox"
                />
              </div>
            </div>
            <label
              for="updater-3"
              id="label-updater-3"
            >
              Auto-updates enabled
            </label>
          </div>
        </div>
        <div
          class="mdc-layout-grid__cell mdc-layout-grid__cell--span-2 mdc-layout-grid__cell--align-middle material-settings__cell--justify-end"
        >
          <button
            class="mdc-button mdc-button--raised"
          >
            <i
              aria-hidden="true"
              class="material-icons mdc-button__icon leading-icon"
            >
              cached
            </i>
            <span
              class="mdc-button__label"
            >
              Update
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Settings: Updater matches snapshot when should not update 1`] = `
<div>
  <div
    class="material-settings__updater"
  >
    <div
      class="mdc-layout-grid"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="mdc-layout-grid__cell mdc-layout-grid__cell--span-7 mdc-layout-grid__cell--align-middle"
        >
          <h3
            class="mdc-typography--headline6"
          >
            Google Fonts
          </h3>
          <p
            class="mdc-typography--body1"
          >
            Last update on Apr 7, 2021, 09:47 PM
          </p>
        </div>
        <div
          class="mdc-layout-grid__cell mdc-layout-grid__cell--span-3 mdc-layout-grid__cell--align-middle material-settings__cell--justify-end"
        >
          <div
            class="material-settings__switch"
          >
            <div
              class="mdc-switch material-wizard-switch mdc-switch--checked"
            >
              <div
                class="mdc-switch__track"
              />
              <div
                class="mdc-switch__thumb-underlay"
              >
                <div
                  class="mdc-switch__thumb"
                />
                <input
                  aria-checked="true"
                  checked=""
                  class="mdc-switch__native-control"
                  id="updater-4"
                  role="switch"
                  type="checkbox"
                />
              </div>
            </div>
            <label
              for="updater-4"
              id="label-updater-4"
            >
              Auto-updates enabled
            </label>
          </div>
        </div>
        <div
          class="mdc-layout-grid__cell mdc-layout-grid__cell--span-2 mdc-layout-grid__cell--align-middle material-settings__cell--justify-end"
        >
          <button
            class="mdc-button mdc-button--raised"
            disabled=""
          >
            <i
              aria-hidden="true"
              class="material-icons mdc-button__icon leading-icon"
            >
              done
            </i>
            <span
              class="mdc-button__label"
            >
              Updated
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
